src/backend/joanie/tests/core/test_models_order.py

```diff
@@ -1007,3 +988,95 @@ def test_models_order_submit_for_signature_check_contract_context_course_section
+        """
+        Check that the `is_free` property returns False if the order total is not 0.
+        """
+        order = factories.OrderFactory(product__price=1)
+        self.assertFalse(order.is_free)
+
+    def test_models_order_has_payment_method(self):
+        """
+        Check that the `has_payment_method` property returns True if the order owner credit
+        card has an initial issuer transaction identifier.
+        """
+        credit_card = CreditCardFactory(
+            initial_issuer_transaction_identifier="4575676657929351"
+        )
+        order = factories.OrderFactory(owner=credit_card.owner)
+        self.assertTrue(order.has_payment_method)
+
+    def test_models_order_has_payment_method_no_transaction_identifier(self):
+        """
+        Check that the `has_payment_method` property returns False if the order owner credit
+        card has no initial issuer transaction identifier.
+        """
+        order = factories.OrderFactory(
+            credit_card=CreditCardFactory(initial_issuer_transaction_identifier=None)
+        )
+        self.assertFalse(order.has_payment_method)
+
+    def test_models_order_has_unsigned_contract(self):
+        """
+        Check that the `has_contract` property returns True if the order has a contract.
+        """
+        order = factories.OrderFactory()
+        factories.ContractFactory(
+            order=order,
+            definition=factories.ContractDefinitionFactory(),
+        )
+        self.assertTrue(order.has_unsigned_contract)
+
+    def test_models_order_has_unsigned_contract_no_contract(self):
+        """
+        Check that the `has_contract` property returns True if the order has a contract.
+        """
+        order = factories.OrderFactory()
+        self.assertFalse(order.has_unsigned_contract)
+
+    def test_models_order_has_unsigned_contract_no_signature(self):
+        """
+        Check that the `has_unsigned_contract` property returns True
+        if the order has an unsigned contract.
+        """
+        order = factories.OrderFactory()
+        factories.ContractFactory(
+            order=order,
+            definition=factories.ContractDefinitionFactory(),
+        )
+        self.assertTrue(order.has_unsigned_contract)
+
+    def test_models_order_has_unsigned_contract_signature(self):
+        """
+        Check that the `has_unsigned_contract` property returns False
+        if the order has an signed contract.
+        """
+        order = factories.OrderFactory()
+        factories.ContractFactory(
+            order=order,
+            definition=factories.ContractDefinitionFactory(),
+            student_signed_on=datetime(2023, 9, 20, 8, 0, tzinfo=timezone.utc),
+            submitted_for_signature_on=datetime(2023, 9, 20, 8, 0, tzinfo=timezone.utc),
+        )
+        self.assertFalse(order.has_unsigned_contract)
+
+    def test_models_order_has_unsigned_contract_false(self):
```

Comment:
It feels like the update was not fully done. The data in the test seems the same in `test_models_order_has_unsigned_contract_signature` test.